core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -409,9 +409,11 @@ public void run() {
                   ManagedClientTransport transport = shutdownDueToUpdateTransport;
                   shutdownDueToUpdateTask = null;
                   shutdownDueToUpdateTransport = null;
-                  transport.shutdown(
-                      Status.UNAVAILABLE.withDescription(
-                          "InternalSubchannel closed transport due to address change"));
+                  if (transport != null) {
```

Comment:
I thought we talked about this one earlier. If we look at every instance of `shutdownDueToUpdateTransport = null`, there's also `shutdownDueToUpdateTask.cancel();`. And when `shutdownDueToUpdateTask` is created, `savedTransport` is guaranteed non-null. I see no other assignments to `shutdownDueToUpdateTransport`.

I don't want to throw random null checks when we have no clue how it'd happen. It's becomes impossible to reason about the code and know what is/is not supported. We should find how it happens and fix it. (The bug is the value is `null`, not that we aren't checking for `null`.) Papering over the symptom just makes the next error with the same cause even harder to debug. For example, here it may leak the transport.

This code is four years old, the failing version was 8 months when we first looked at it, and now it is 14. This file wasn't changed between the working and failing version. No other instance of it being reported, even though it is a panic. Right now I'm willing to blame cosmic rays.